src/ethereum/cancun/vm/__init__.py

```diff
@@ -57,6 +70,8 @@ class Message:
     Items that are used by contract creation or message call.
     """
 
+    block_env: BlockEnvironment
+    tx_env: TransactionEnvironment
```

Comment:
Would it make sense to introduce a third "environment" for the frame?

```python
class BlockEnvironment:
    ...

class TransactionEnvironment:
    ...

class FrameEnvironment:
    caller: Address
    target: Union[Bytes0, Address]
    current_target: Address
    gas: Uint
    value: U256
    data: Bytes
    code_address: Optional[Address]
    code: Bytes
    depth: Uint
    should_transfer_value: bool
    is_static: bool
    accessed_addresses: Set[Address]
    accessed_storage_keys: Set[Tuple[Address, Bytes32]]
    parent_evm: Optional["Evm"]

class Message:
    block: BlockEnvironment
    transaction: TransactionEnvironment
    frame: FrameEnvironment
```

Reply:
This would line up the nomenclature pretty well but I have been looking at the `Message` dataclass serving the same purpose as the `FrameEnvironment` dataclass in your suggestion. I don't have any strong opinions on either choice tbh.

Reply:
The symmetry argument sounds good to me but the question largely boils down to complexity since introducing `FrameEnvironment` could be a large change. Also, I think we can introduce `FrameEnvironment` in the next iteration to avoid a huge PR

Reply:
I think that makes sense.